src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/Contracts/ExternalLanguageModel.cs

```diff
@@ -0,0 +1,116 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.SqlTools.ServiceLayer.LanguageExtensibility.Contracts
+{
+
+    public enum ExternalLanguagePlatform
+    {
+        None,
+        Windows,
+        Linux
+    }
+
+    /// <summary>
+    /// Language metadata
+    /// </summary>
+    public class ExternalLanguage
+    {
+
+        /// <summary>
+        /// Language Name
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Language Owner
+        /// </summary>
+        public string Owner { get; set; }
+
+        public List<ExternalLanguageContent> Contents { get; set; }
+
+        /// <summary>
+        /// Created Date
+        /// </summary>
+        public string CreatedDate { get; set; }
+
+    }
+
+    public class ExternalLanguageContent
+    {
+        public bool IsLocalFile { get; set; }
+
+        /// <summary>
+        /// Path to extension file
+        /// </summary>
+        public string PathToExtension { get; set; }
+
+        public object Content
+        {
+            get
+            {
+                if (IsLocalFile)
+                {
+                    return ExtensionFileBytes;
+                }
+                else
+                {
+                    return PathToExtension;
+                }
+            }
+        }
+
+        public byte[] ExtensionFileBytes
+        {
+            get
+            {
+                if (IsLocalFile)
+                {
+                    using (var stream = new FileStream(PathToExtension, FileMode.Open, FileAccess.Read))
```

Comment:
the contacts are supposed to define the wire-format for JSON-RPC communications.  It seems unusual to have this type of logic in a getter in a contact.  Perhaps this would be better in a utility class in the actual applogic implementation?

Reply:
sure I move it to the service code, not needed in the contract 

---
In reply to: [379156888](https://github.com/microsoft/sqltoolsservice/pull/918#discussion_r379156888) [](ancestors = 379156888)